### PR TITLE
Handle ABAP Language Version for source code objects

### LINKED ISSUE #6426: Consistent ABAP Language Version for CLAS, INTF, ...
With this issue, we track the progress for the adoption of a consistent ABAP Language Version (ALaV) for:

- [x]  Class 
  (contains `<UNICODE>X</UNICODE>`)
- [x]  Interfaces 
  (contains `<UNICODE>X</UNICODE>`)
- [x]  Function Groups 
  (contains `<UCCHECK>X</UCCHECK>`)
- [x]  Programs 
  (contains `<UCCHECK>X</UCCHECK>`)
- [x] Packages
  (can contain `<PACKAGE_KIND>5</PACKAGE_KIND>`)
- [x]  Enhancements (does not contain any ALaV flag but needs to be tested)

Use the functionality introduced in #6346 

Expectation when using ALaV feature:
- Result of serialize should not contain ALaV (i.e. the above data)
- Deserialize should set ALaV according to agreed rules

### PR DESCRIPTION
**Pull request by mbtools**
Introduce experimental feature to handle ABAP Language Version for `PROG`, `FUGR`, `CLAS`, and `INTF`. 

Feature flag for global settings: `ALAV`

When feature is enabled, the XML files do not contain any `UCCHECK` or `UNICODE` fields. These fields are derived from the SAP package of the repo during deserialize.

Test case: https://github.com/abapGit-tests/DDIC_integration

- `main` branch contains the objects serialized as usual
- `abap_standard` branch contains repo setting = `standard` 
- `abap_cloud_development` branch contains repo setting = `cloud_development` 

Test flow:

- Create repo from `main` branch and pull 
- Uninstall repo
- Enable ALAV feature
- Create repo from `abap_standard` or `abap_cloud_development` branch and pull
- Uninstall repo
- Disable ALAV feature
 
Closes #6426
Ref #6154, #6298

### CODE REVIEW

**Comment by albertmink** [comment]
I cannot test this one, see #6477

**Comment by mbtools** [comment]
I reviewed #6346 again and this PR is deriving ALaV unnecessarily twice. Will adjust

**Comment by mbtools** [comment]
Will keep open but splitting into smaller PRs:
https://github.com/abapGit/abapGit/pull/6484

**Comment by mbtools** [comment]
- Passed all CI tests (with ALAV feature off) 😃 
- Lots of "observations". Will address this separately so as to not make this messy
- Add ALAV tests to CI (https://github.com/abapGit/CI/issues/239)

**Comment by schneidermic0** [comment]
@mbtools If you like I can review the changes. Unfortunately, I guess I won't manage it today.

**Comment by larshp** [comment]
> Lots of "observations". Will address this separately so as to not make this messy

exactly, leave those as is. This change is important/complex so no need to include those here, there will be another chance with something more simple changes in the future

**Comment by mbtools** [comment]
any time @schneidermic0. note, as you can see, the logic is implemented for source code objects only. for others like DDIC, the "set_abap_language_version" still needs to be added. do you have a list of object types that support ALAV to cross-reference with abapGit?

**Comment by schneidermic0** [comment]
> do you have a list of object types that support ALAV to cross-reference with abapGit?

I'll check

**Comment by schneidermic0** [comment]
> do you have a list of object types that support ALAV to cross-reference with abapGit?

I have just added the object types I found to https://github.com/abapGit/abapGit/issues/6154#issuecomment-1749086748

**Review by schneidermic0 — COMMENTED** [review]
In general, I think the changes look good.

I added just one small comment.

File: src/objects/zcl_abapgit_object_type.clas.abap [hunk 1]
```diff
@@ -143,6 +143,7 @@ CLASS zcl_abapgit_object_type IMPLEMENTATION.
       zcl_abapgit_factory=>get_sap_report( )->insert_report(
         iv_name    = lv_progname
         iv_package = iv_package
+        iv_version = abap_true
```
[the remark this answers is not in the document]
Comment (mbtools) [reply src/objects/zcl_abapgit_object_type.clas.abap]: lazy me 😉. I changed to a constant

**Pull request merged**